validator/src/main/java/com/amazon/aoc/validators/ContainerInsightStructuredLogValidatorContainerd.java

```diff
@@ -0,0 +1,94 @@
+package com.amazon.aoc.validators;
+
+import com.amazon.aoc.fileconfigs.FileConfig;
+import com.amazon.aoc.fileconfigs.LocalPathExpectedTemplate;
+import com.amazon.aoc.helpers.MustacheHelper;
+import com.amazon.aoc.models.Context;
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.services.logs.model.FilteredLogEvent;
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+
+@Log4j2
+public class ContainerInsightStructuredLogValidatorContainerd
+        extends AbstractStructuredLogValidator {
+
+  //expected template Array for containerd runtime
+  private static final List<String> LOG_TYPE_CONTAINERD = Arrays.asList(
+          "Cluster",
+          "ClusterNamespace",
+          "ClusterService",
+          "Container",
+          "Node",
+          "NodeDiskIO",
+          "NodeFS",
+          "NodeNet",
+          "Pod",
+          "PodNet"
+  );
+
+  private static final int MAX_RETRY_COUNT = 15;
+  private static final int QUERY_LIMIT = 100;
+
+  @Override
+  void init(Context context, FileConfig expectedDataTemplate) throws Exception {
+    logGroupName = String.format("/aws/containerinsights/%s/performance",
+            context.getCloudWatchContext().getClusterName());
+    MustacheHelper mustacheHelper = new MustacheHelper();
+    for (String logType : LOG_TYPE_CONTAINERD) {
+      FileConfig fileConfig = new LocalPathExpectedTemplate(FilenameUtils.concat(
+              expectedDataTemplate.getPath().toString(),
+              logType + ".json"));
+      try {
+        String templateInput = mustacheHelper.render(fileConfig, context);
+        schemasToValidate.put(logType, parseJsonSchema(templateInput));
+      } catch (IOException e) {
+        log.debug("The " + logType + " was not found for this expected template path.");
+      }
+    }
+  }
+
+  @Override
+  String getJsonSchemaMappingKey(JsonNode jsonNode) {
+    return jsonNode.get("Type").asText();
+  }
+
+  @Override
+  protected int getMaxRetryCount() {
+    return MAX_RETRY_COUNT;
+  }
+
+  @Override
+  protected void fetchAndValidateLogs(Instant startTime) throws Exception {
```

Comment:
Is this duplicate code from  `ContainerInsightStructuredLogValidatorDocker`? Can we store this in an abstract class that `ContainerInsightStructuredLogValidatorDocker` and `ContainerInsightStructuredLogValidatorContainerd` extend?

Reply:
I think is similar to what @rapphil has requested here https://github.com/aws-observability/aws-otel-test-framework/pull/1118#discussion_r1139093992. We should have a base class for this.